proto/encore/parser/schema/v1/schema.proto

```diff
@@ -4,88 +4,141 @@ option go_package = "encr.dev/proto/encore/parser/schema/v1";
+// replaced with a type argument upon instantiation of the parameterized function or type.
+message TypeParameter {
+  string name = 1; // The identifier given to the type parameter
+}
+
+// Loc is the location of a declaration within the code base
 message Loc {
-  string pkg_path = 1;
-  string pkg_name = 2;
-  string filename = 3;
-  int32 start_pos = 4;
-  int32 end_pos = 5;
-  int32 src_line_start = 6;
-  int32 src_line_end = 7;
-  int32 src_col_start = 8;
-  int32 src_col_end = 9;
+  string pkg_path       = 1; // The package path within the repo (i.e. `/users/signup`)
+  string pkg_name       = 2; // The package name (i.e. `signup`)
+  string filename       = 3; // The file name (i.e. `signup.go`)
+  int32  start_pos      = 4; // The starting index within the file for this node
+  int32  end_pos        = 5; // The ending index within the file for this node
+  int32  src_line_start = 6; // The starting line within the file for this node
+  int32  src_line_end   = 7; // The ending line within the file for this node
+  int32  src_col_start  = 8; // The starting column on the starting line for this node
+  int32  src_col_end    = 9; // The ending column on the ending line for this node
 }
 
+// Named references declaration block by name
 message Named {
-  uint32 id = 1;
+  uint32   id                  = 1; // The `Decl.id` this name refers to
+  repeated Type type_arguments = 2; // The type arguments used to instantiate this parameterised declaration
 }
 
+// Struct contains a list of fields which make up the struct
 message Struct {
   repeated Field fields = 1;
 }
 
+// Field represents a field within a struct
 message Field {
-  Type typ = 1;
-  string name = 2;
-  string doc = 3;
-  // The optional json name if it's different from the field name.
-  // The value "-" indicates to omit the field.
-  string json_name = 4;
-  // Whether the field is optional.
-  bool optional = 5;
-  // The query string name to use in GET/HEAD/DELETE requests.
-  // The value "-" indicates to omit the field.
-  string query_string_name = 6;
+  Type   typ               = 1; // The type of the field
+  string name              = 2; // The name of the field
+  string doc               = 3; // The comment for the field
+  string json_name         = 4; // The optional json name if it's different from the field name. (The value "-" indicates to omit the field.)
+  bool   optional          = 5; // Whether the field is optional.
+  string query_string_name = 6; // The query string name to use in GET/HEAD/DELETE requests. (The value "-" indicates to omit the field.)
 }
 
+// Map represents a map Type
 message Map {
-  Type key = 1;
-  Type value = 2;
+  Type key   = 1; // The type of the key for this map
+  Type value = 2; // The type of the value of this map
 }
 
+// List represents a list type (array or slice)
 message List {
-  Type elem = 1;
+  Type elem = 1; // The type of the list
```

Comment:
Also seems a bit redundant, but if we want to keep it it should rather say "The type of the elements in the list"